portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -14,14 +14,31 @@
 
 package com.google.sps.data;
 
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+
 /** Class to store user's website comments. */
 public final class Comment {
 
   private final String message;
   private final String userData;
+  private final float score;
 
-  public Comment (String message, String userData) {
+  public Comment(String message, String userData) throws IOException {
     this.message = message;
     this.userData = userData;
+    this.score = getMessageScore(message);
+  }
+
+  private float getMessageScore(String message) throws IOException {
+    Document doc = Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    float score = sentiment.getScore();
```

Comment:
The dance you do here around this variable seems to imply that `sentiment` becomes invalid after `languageService.close` is called.

But I'm not quite sure why that would be the case. Would something like this not work?

```
client = LanguageServiceClient.create();
sentiment = ...getDocumentSentiment();
client.close();

return sentiment.getScore();
```

Reply:
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/language/analysis/src/main/java/com/google/cloud/language/samples/Analyze.java#L160

suggests using the try-with-resource format here (so that you don't have to worry about properly closing), and handling the possibility of sentiment being null.

Reply:
I changed the code to use try-with-resource, but the version you mentioned (which closes the client before returning the sentiment score) would also work: